### ISSUE TITLE
Simple C ChaCha20 and Poly1305 for IETF  (AEAD)

### ISSUE THREAD

**Issue by daviddpd** [daviddpd/dpdChatFabric#1]
I can't seem to find a simple ChaCha20 C implementation for RFC 7539.  Most are 64bit nonce , the IETF version uses a 96 bit nonce.

Currently, encrypted payloads are Salsa20 + Poly1305.

**Comment by beckmx** [comment]
so, does it mean the esp8266 might not be able to implement chacha20?

**Comment by daviddpd** [comment]
Not at all. I just can't find C99 code for chacha20 (RFC 7539) w/ the 96 bin nonce.  The encryption code is just a little bit outside my skill and patience level to implement. 

There is absolutely not issue with esp8266 cpu power for encryption.

**Comment by pallas** [comment]
https://github.com/wg/c20p1305
